flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -220,7 +220,7 @@ INTERNAL_HADOOP_CLASSPATHS="$HADOOP_CLASSPATH:$HADOOP_CONF_DIR:$YARN_CONF_DIR"
 # also potentialy includes topology information and the taskManager type
 extractHostName() {
     # extract first part of string (before any whitespace characters)
-    SLAVE=$1
+    SLAVE=`echo $1 | cut -d'#' -f 1`
     # Remove types and possible comments
     if [[ "$SLAVE" =~ ^([0-9a-zA-Z/.-]+).*$ ]]; then
             SLAVE=${BASH_REMATCH[1]}
```

Comment:
No need to change line 223 because
```bash
if [[ "$SLAVE" =~ ^([0-9a-zA-Z/.-]+).*$ ]]; then
             SLAVE=${BASH_REMATCH[1]}
fi
``` 
will already filter out lines starting with `#`.